cmd/celestia-appd/cmd/modify_root_command_multiplexer.go

```diff
@@ -14,6 +14,13 @@ import (
 	embedding "github.com/celestiaorg/celestia-app/v4/internal/embedding"
 )
 
+// v2UpgradeHeight is the block height at which the v2 upgrade occurred.
+// this can be overridden at build time using ldflags:
+// -ldflags="-X 'github.com/celestiaorg/celestia-app/v4/cmd/celestia-appd/cmd.v2UpgradeHeight=1751707'" for arabica
+// -ldflags="-X 'github.com/celestiaorg/celestia-app/v4/cmd/celestia-appd/cmd.v2UpgradeHeight=2585031'" for mocha
+// -ldflags="-X 'github.com/celestiaorg/celestia-app/v4/cmd/celestia-appd/cmd.v2UpgradeHeight=2371495'" for mainnet
+var v2UpgradeHeight = "0"
```

Comment:
If the user doesn't override this value then `0` will be plumbed into `--v2-upgrade-height` which will take precedence over the defaults: https://github.com/celestiaorg/celestia-app/pull/4607 so we need to invoke v3 binary without `--v2-upgrade-height` if v2UpgradeHeight is `0`

Reply:
Gotcha. I see the pr sets it as default indeed and doesn't override it (SetDefault instead of Set). Was there a reason for that?

Reply:
Hmm I think I used `SetDefault` so that any value specified by user via `--v2-upgrade-height` would take precedence.

Not strongly opinionated on how we do it. We can modify v3.x to always use the hard-coded upgrade heights for the supported networks if that helps this PR.

Reply:
Nah, it's easier to skip here. That's fine. I'll update in a bit

Reply:
fixed